stdlib/test/collections/test_deque.mojo

```diff
@@ -355,6 +554,112 @@ fn test_getitem() raises:
     assert_equal(q[-2], 1)
 
 
+fn test_setitem() raises:
+    q = Deque(1, 2)
+    assert_equal(q[0], 1)
+
+    q[0] = 3
+    assert_equal(q[0], 3)
+
+    q[-1] = 4
+    assert_equal(q[1], 4)
+
+
+fn test_eq() raises:
+    q = Deque[Int](1, 2, 3)
+    p = Deque[Int](1, 2, 3)
+
+    assert_true(q == p)
+
+    r = Deque[Int](0, 1, 2, 3)
+    q.appendleft(0)
+    assert_true(q == r)
+
+
+fn test_ne() raises:
+    q = Deque[Int](1, 2, 3)
+    p = Deque[Int](3, 2, 1)
+
+    assert_true(q != p)
+
+    q.appendleft(0)
+    p.append(0)
+    assert_true(q != p)
+
```

Comment:
**Question** What's the behavior/do you mind adding a test for when the element types differ but the values are the same?  Does that even compile currently?  E.g.

```
q = Deque[Int](1, 2, 3)
p = Deque[UInt](1, 2, 3)
assert_true(q != p) # what's the behavior here?
```

Reply:
No, it does not compile. Neither this compiles with `List`.
```
err.mojo:7:13: error: invalid call to '__eq__': could not deduce positional-only parameter #1 of callee '__eq__'
    print(q == p)
          ~~^~~~
err.mojo:7:16: note: failed to infer parameter #1, parameter inferred to two different values: 'Int' and 'UInt'
    print(q == p)
               ^
deque.mojo:267:8: note: function declared here
        EqualityElementType: EqualityComparableCollectionElement, //
       ^
mojo: error: failed to parse the provided Mojo source module
```
As both arguments of `__eq__` are defined being of type `EqualityElementType`, I suppose they are expected to be of exactly the same type.

Reply:
Great, that's exactly what I expected to happen - thanks for confirming.